examples/pennylane_run_isingmodel_PyTorch.py

```diff
@@ -0,0 +1,151 @@
+# Random initialization
+# ^^^^^^^^^^^^^^^^^^^^^
+
+torch.manual_seed(56)
+p1 = Variable((np.pi * torch.rand(3, dtype=torch.float64)), requires_grad=True)
+p2 = Variable((np.pi * torch.rand(3, dtype=torch.float64)), requires_grad=True)
+
+var_init = [p1, p2]
+cost_init = cost(p1, p2)
+
+print("Randomly initialized angles:",var_init)
+print("Corresponding cost before optimization:",cost_init)
+
+###############################################################################
+# Optimization
+# ^^^^^^^^^^^^
+# Now we use the PyTorch gradient descent optimizer to minimize the cost:
+
+opt = torch.optim.SGD(var_init, lr = 0.1)
+
+steps = 100
+
+def closure():
+    opt.zero_grad()
+    loss = cost(p1, p2)
+    loss.backward()
+    return loss
+
+var_pt = [var_init]
+cost_pt = [cost_init]
+
+for i in range(steps):
+    opt.step(closure)
+    if (i + 1) % 5 == 0:
+        p1n, p2n = opt.param_groups[0]["params"]
+        costn = cost(p1n, p2n)
+        var_pt.append([p1n, p2n])
+        cost_pt.append(costn)
+        print("Energy after step {:5d}: {: .7f} | Angles: {}".format(i + 1, costn, [p1n, p2n]),"\n")
+
+###############################################################################
+#
+# .. note::
+#     When using the *PyTorch* optimizer, keep in mind that:
+#
+#     1. ``loss.backward()`` computes the gradient of the cost function with respect to all parameters (whcih have ``requires_grad=True``). 
+#     2. ``opt.step()`` performs the parameter update based on this *current* gradient and the learning rate. 
+#     3. ``opt.zer_grad()`` sets all the gradients back to zero. It’s important to call this before ``loss.backward()`` to avoid the accumulation of  gradients from multiple passes.
+#
+#     Hence, its standard practice to define the ``closure()`` function that clears up the old gradient, 
+#     evaluates the new gradient and passes it onto the optimizer in each step. 
+#
+# The minimum energy is -2  for the spin configuration :math:`[s_1, s_2, s_3] = [1, 1, -1]`
+# which corresponds to
+# :math:`(\phi, \theta, \omega) = (0, 0, 0)` for the second spin and :math:`(\phi, \theta, \omega) = (0, \pi, 0)` for 
+# the third spin. We might not always see this cost value due to the non-convex cost function.
+
+p1_final, p2_final = opt.param_groups[0]["params"]
+print("Optimized angles:",p1_final, p2_final)
+print("Final cost after optimization:",cost(p1_final, p2_final))
+
+###############################################################################
+#
+# Local minimum
+# ^^^^^^^^^^^^^
+# If the spins are initialized close to the local minimum of zero energy, the optimizer is
+# likely to get stuck here and never find the global minimum at -2. The figure below shows
+# the results from two different initializations on various optimizers.
+# 
+# |
+# .. image:: ../../examples/figures/ising1.png
+#    :width: 48%
+# .. image:: ../../examples/figures/ising2.png
+#    :width: 48%
```

Comment:
It might be a bit confusing that the image shows a TensorFlow optimization and a numpy/autograd optimization without the code/context appearing in the tutorial above. Do you think it's worth replacing this with a plot of just the above pytorch cost vs. step?

Reply:
Hm I think it shows consistency across optimizers but If you think it is confusing I can replace it with a plot with just Pytorch.

Reply:
Hm I think it shows consistency across optimizers but If you think it is confusing I can replace it with a plot with just Pytorch.